eth-registration/src/main/kotlin/com/d3/eth/registration/EthRegistrationStrategyImpl.kt

```diff
@@ -47,33 +55,51 @@ class EthRegistrationStrategyImpl(
         ethRegistrationConfig.ethRelayRegistryAddress,
         web3,
         credentials,
-        BigInteger.valueOf(ethRegistrationConfig.ethereum.gasPrice),
-        BigInteger.valueOf(ethRegistrationConfig.ethereum.gasLimit)
+        StaticGasProvider(
+            BigInteger.valueOf(ethRegistrationConfig.ethereum.gasPrice),
+            BigInteger.valueOf(ethRegistrationConfig.ethereum.gasLimit)
+        )
     )
 
     private val irohaEthAccountCreator =
         IrohaEthAccountCreator(irohaConsumer, notaryIrohaAccount)
 
     /**
      * Register new notary client
-     * @param name - client name
+     * @param accountName - client name
+     * @param domainId - client domain
      * @param pubkey - client public key
      * @param whitelist - list of addresses from client
      * @return ethereum wallet has been registered
      */
     override fun register(
-        name: String,
-        domain: String,
+        accountName: String,
+        domainId: String,
         whitelist: List<String>,
         pubkey: String
     ): Result<String, Exception> {
         return ethFreeRelayProvider.getRelay()
             .flatMap { freeEthWallet ->
-                logger.info { "Add new relay to relay registry relayRegistry=$relayRegistry, freeWallet=$freeEthWallet, whitelist=$whitelist, creator=${credentials.address}." }
-                relayRegistry.addNewRelayAddress(freeEthWallet, whitelist).send()
-                irohaEthAccountCreator.create(
-                    freeEthWallet, whitelist, name, domain, pubkey
-                )
+                ethRelayProvider.getRelaysByAccountId("$accountName@$domainId")
+                    .flatMap { assignedRelays ->
+                        // check that client hasn't been registered yet
+                        if (!assignedRelays.isEmpty())
+                            throw IllegalArgumentException("Client $accountName@$domainId has already been registered with relay: $assignedRelays")
+
+                        // register to Ethereum RelayRegistry
+                        logger.info { "Add new relay to relay registry relayRegistry=$relayRegistry, freeWallet=$freeEthWallet, whitelist=$whitelist, creator=${credentials.address}." }
+                        relayRegistry.addNewRelayAddress(freeEthWallet, whitelist).send()
+
+                        // register to Iroha
+                        irohaAccountCreator.create(
```

Comment:
User should have account in Iroha in order to register in Ethereum sidechain.
Updated comment.

Reply:
Sure, but I'd also recommend to rename `create` and `createAccountCreationBatch` methods to avoid confusion. For example, we could use some sort of `set registered in side chain information in Iroha` terminology.